posthog/year_in_posthog/calculate_2022.py

```diff
@@ -0,0 +1,178 @@
+from datetime import timedelta
+from typing import Dict, Optional
+
+from django.db import connection
+
+from posthog.cache_utils import cache_for
+from posthog.logging.timing import timed
+
+query = """
```

Comment:
Have you tried this in prod? We have quite strict timeouts in prod pg meaning if it will time out then it will likely time out.

Reply:
It's really fast (in metabase). Sub one second.

I had a previous PR that stored state assuming it'd be slow enough to be a problem.

Reply:
Nice! You can disregard my concerns around caching then :)